src/django_components/extension.py

```diff
@@ -255,18 +256,25 @@ class Cache(CacheExtension.ComponentConfig):
     component_class: Type["Component"]
     """The [`Component`](./api.md#django_components.Component) class that this extension is defined on."""
 
-    component: "Component"
-    """
-    When a [`Component`](./api.md#django_components.Component) is instantiated,
-    also the nested extension classes (such as `Component.View`) are instantiated,
-    receiving the component instance as an argument.
+    @property
+    def component(self) -> "Component":
+        """
+        When a [`Component`](./api.md#django_components.Component) is instantiated,
+        also the nested extension classes (such as `Component.View`) are instantiated,
+        receiving the component instance as an argument.
 
-    This attribute holds the owner [`Component`](./api.md#django_components.Component) instance
-    that this extension is defined on.
-    """
+        This attribute holds the owner [`Component`](./api.md#django_components.Component) instance
+        that this extension is defined on.
+        """
+        component = self._component_ref()
+        if component is None:
+            raise RuntimeError("Component has been garbage collected")
+        return component
 
     def __init__(self, component: "Component") -> None:
-        self.component = component
```

Comment:
Each extension has access to the Component instance. But this prevented Components from being garbage collected. So now extensions keep only a weak ref to the component.